go/database/mpt/verification.go

```diff
@@ -570,6 +612,61 @@ func verifyHashesStoredWithParents[N any](
 	return nil
 }
 
+// verifyContractCodes runs list of validation checks on accounts and on the
+// code file stored in the given directory. These checks include:
+// 1) Fatal checks
+// - All CodeHashes within accounts are present in the code file
+// - All CodeHashes within the code file are correct matching the contract byte-codes
+// 2) Non-fatal checks
+// - There are no extra Code Hashes not referenced by any account
+func verifyContractCodes(directory string, source *verificationNodeSource, observer VerificationObserver) error {
+	observer.Progress(fmt.Sprintf("Checking contract codes ..."))
+
+	codeFile := filepath.Join(directory, "codes.dat")
+	codes, err := readCodes(codeFile)
+	if err != nil {
+		return err
+	}
+
+	checkedHashes := make(map[common.Hash]bool)
+	check := func(acc *AccountNode) error {
+		codeHash := acc.info.CodeHash
+		// skip accounts that are not contracts
+		if codeHash == emptyCodeHash {
+			return nil
+		}
+		// no need to check the hash correctness more than once
+		if _, isChecked := checkedHashes[codeHash]; isChecked {
```

Comment:
```suggestion
		if _, exists := checkedHashes[codeHash]; exists {
```

Reply:
I think it is standard to use either comma-ok or comma-exists idioms. It is not necessary to be too creative in naming